apps/emqx/src/emqx_config.erl

```diff
@@ -911,3 +914,12 @@ rawconf_to_conf(SchemaModule, RawPath, RawValue) ->
         ),
     AtomPath = to_atom_conf_path(RawPath, {raise_error, maybe_update_zone_error}),
     emqx_utils_maps:deep_get(AtomPath, RawUserDefinedValues).
+
+%% When the global zone change, the zones is updated with the new global zone.
+%% The global zone's keys is too many,
+%% so we don't choose to write a global zone change emqx_config_handler callback to hook
+post_save_config_hook(?PERSIS_KEY(?CONF, zones), _Zones) ->
```

Comment:
this is not a hook, at least not the current form.
maybe name it `zone_config_post_change_actions`